### ISSUE TITLE
Long tokens with internal newlines are missed by Scanner

### ISSUE THREAD

**Issue by no-reply** [gkellogg/ebnf#5]
```ruby
str = "+#{'0' * 10000}\n+"
s = EBNF::LL1::Scanner.new(StringIO.new(str))
s.scan(/\+.*\+/m)
# => nil
```

Increasing *both* `:low_water` and `:high_water` to a size larger than the token fixes this.

```ruby
s = EBNF::LL1::Scanner.new(StringIO.new(str), :low_water => 20000, :high_water => 20000)
s.scan(/\+.*\+/m)
```

When increasing only `:high_water`, there's a possibility that the previous scan/feed will leave us with a `#rest` containing only part of the next token.

I'm not sure of a way to fix this, beyond giving up on low/high water scanning altogether; it seems like this would be a problem for any language with arbitrarily long multiline terminals.

**Comment by gkellogg** [comment]
It should be possible to see that the terminal has not been matched and continue loading until it is. But, increasing the limits through options passed rom the Turtle parser would be a short-term workaround.

**Comment by gkellogg** [comment]
We could add terminal definition option for the long strings to cause them to keep loading.mthis still leaves the potential for loading the rest of the file, if the literal is poorly formed, but any processor would have the same problem.

**Comment by no-reply** [comment]
I believe this issue is fully closed by 782aead.
